firmware/thruna/BMS/Src/App/states/App_AllStates.c

```diff
@@ -40,7 +48,10 @@ static bool App_SendAndReceiveHeartbeat(struct HeartbeatMonitor *hb_monitor)
         App_CanRx_PDM_Heartbeat_Update(false);
     }
 
-    const bool missing_hb = !App_SharedHeartbeatMonitor_Tick(hb_monitor);
+    const enum HeartbeatOneHot hb_state = App_SharedHeartbeatMonitor_Tick(hb_monitor);
+    App_SharedHeartbeatMonitor_BroadcastFaults(hb_state, heartbeatCanSetters, true);
+
+    const bool missing_hb = !App_SharedHeartbeatMonitor_ValidateState(hb_state);
```

Comment:
Hmm I was hoping that the previous lines where we get the heartbeats and check them in could be handled by the heartbeat driver itself
And then we only have to call the tick function from the state machine